objects.go

```diff
@@ -261,6 +350,25 @@ func (set *Set) writeOperation(verb verb, ctx *nftContext, writer io.Writer) {
 	fmt.Fprintf(writer, "\n")
 }
 
+func (set *Set) parse(line string) error {
+	match := setRegexp.FindStringSubmatch(line)
+	if match == nil {
+		return fmt.Errorf("failed parsing set add command")
+	}
+	name, typeProp, typeOf, flags, timeout, gcInterval, size, policy, comment, autoMerge := parseMapAndSetProps(match)
+	set.Name = name
+	set.Type = typeProp
```

Comment:
you don't need the temporary variables, you could just do

    set.Name, set.Type, set.TypeOf, (...) = parseMapAndSetProps(match)

(maybe with a line break or two)